### PR TITLE
git fetch support if the git directory already exists

### PR DESCRIPTION
**Pull request by birmacher**

### CODE REVIEW

File: go/src/github.com/bitrise-io/git-clone/gitutil/gitutil.go [hunk 1]
```diff
@@ -138,13 +155,36 @@ func (helper Helper) Init() error {
...
 
+// Clean
+func (helper Helper) Clean() error {
+	cmdSlice := createGitCmdSlice("reset", "--hard", "HEAD")
+	if err := runCommandInDir(cmdSlice, helper.destinationDir); err != nil {
+		return err
+	}
+
+	cmdSlice := createGitCmdSlice("clean", "-f")
+	if err := runCommandInDir(cmdSlice, helper.destinationDir); err != nil {
```
Comment (godrei) [inline go/src/github.com/bitrise-io/git-clone/gitutil/gitutil.go]: `return runCommandInDir(cmdSlice, helper.destinationDir)`

File: go/src/github.com/bitrise-io/git-clone/gitutil/gitutil.go [hunk 2]
```diff
@@ -138,13 +155,36 @@ func (helper Helper) Init() error {
...
 }
 
+// Clean
+func (helper Helper) Clean() error {
+	cmdSlice := createGitCmdSlice("reset", "--hard", "HEAD")
+	if err := runCommandInDir(cmdSlice, helper.destinationDir); err != nil {
+		return err
+	}
+
+	cmdSlice := createGitCmdSlice("clean", "-f")
```
Comment (godrei) [inline go/src/github.com/bitrise-io/git-clone/gitutil/gitutil.go]: `no new variables on left side of :=` - that's why bitrise ci fails

File: go/src/github.com/bitrise-io/git-clone/gitutil/gitutil.go [hunk 3]
```diff
@@ -28,10 +28,11 @@ type Helper struct {
 	checkoutTag   bool
 	pullRequestID string
 	cloneDepth    string
+	originPresent bool
 }
 
 // NewHelper ...
-func NewHelper(destinationDir, remoteURI string) (Helper, error) {
+func NewHelper(destinationDir, remoteURI string, resetRepository string) (Helper, error) {
```
Comment (godrei) [inline go/src/github.com/bitrise-io/git-clone/gitutil/gitutil.go]: `resetRepository` should be boolean

**Comment by birmacher** [comment]
will solve #14

**Comment by godrei** [comment]
👍

**Pull request merged**